File: Python/CaesarCipher.py
##
## Name: Caleb Rogers
## CaesarCipher.py
##
## Purpose:
## Caesar Cipher that utilizes shift variable that determines the amount that characters are shifted.
##
## Inputs:
## 1. String to be encrypted
## 2. Number to shift with
##
## Output:
## Encrypted string
##
## Reference:
## https://www.tutorialspoint.com/cryptography_with_python/cryptography_with_python_caesar_cipher.htm
## https://www.geeksforgeeks.org/caesar-cipher-in-cryptography/
##

# Encryption Phase with shift n:
# E_n(x)=(x+n)mod\ 26

# Decryption Phase with shift n:
# D_n(x)=(x-n)mod\ 26


def main():
    print()
    print("Welcome to Caesar Ciphers with Python!!")
    originalTxt = input("Enter a string to be encrypted: ")
    shift = input("Enter a number to shift with: ")
    encryptedTxt = encrypt(originalTxt, shift)

    print("Plain Text : " + originalTxt)
    print("Shift pattern : " + shift)
    print("Cipher: " + encryptedTxt)


def encrypt(text,s):
    result = ""
    # transverse the plain text
    for i in range(len(text)):
        char = text[i]
        # Encrypt uppercase characters in plain text
        
        if (char.isupper()):
            result += chr((ord(char) + int(s)-65) % 26 + 65)
        # Encrypt lowercase characters in plain text
        else:
            result += chr((ord(char) + int(s) - 97) % 26 + 97)
    return result


main()

File: Scala/caesar_cipher.scala
object CaesarCipher {
    def encrypt(toCipher: String, shift: Int): String = {
		var ciphered: String = "";
		
        for(i <- 0 to (toCipher.length() - 1)) {
                if (toCipher.charAt(i) >= 'A' && toCipher.charAt(i) <= 'Z') {
                    if(!((toCipher.charAt(i)).isWhitespace)) {
                        ciphered = ciphered + (((toCipher.charAt(i).toInt + shift - 65) % 26).toInt + 65).toChar.toString();
                    }
                    else {
                        ciphered = ciphered + toCipher.charAt(i);
                    }
                }
                else {
                    if(!((toCipher.charAt(i)).isWhitespace)) {
                        ciphered = ciphered + (((toCipher.charAt(i).toInt + shift - 97) % 26).toInt + 97).toChar.toString();
                    }
                    else {
                        ciphered = ciphered + toCipher.charAt(i);
                    }
                }
		}
		return ciphered;
	}


    def decrypt(ciphered: String, shift: Int): String = {
        var negativeShift: Int = (26 - shift)
        return encrypt(ciphered, negativeShift)
    }


    def solve(cipherStr: String, loop: Int): Unit = {
        for(i <- 0 to loop) {
            println("Caesar " + i + ": " + encrypt(cipherStr, i))
        }
    }
    
    def main(args: Array[String]): Unit = {
        
        // Initialization
        var cipherStr = "Dude Wheres My Car"
        var shift = 4
        
        println("Welcome to Caesar Ciphers with Scala!")
        
        // Call Caesar Cipher Encryption Method
        var ciphered = encrypt(cipherStr, shift)
        println("Encrypted Cipher Value: " + ciphered)
  
        // Call Caesar Cipher Decryption Method
        var deciphered = decrypt(ciphered, shift)
        println("Decrypted Cipher Value: " + deciphered)
      
        // Call Caesar Cipher Solve Method
        solve("DUDE", 26)
    }
}
